src/xrCore/Crypto/xr_sha.cpp

```diff
@@ -1,24 +1,32 @@
 #include "stdafx.h"
 #include "xr_sha.h"
 
-namespace crypto
-{
+using namespace crypto;
 
-const xr_sha1::sha_checksum_t& xr_sha1::calculate(const u8* data, u32 data_size)
+const xr_sha1::sha_checksum_t& xr_sha1::calculate(const u8* data, u32 data_size, std::optional<sha_process_yielder> yielder)
 {
     VERIFY(data_size);
     const u8* data_pos = data;
     CryptoPP::SHA1 sha_ctx{};
-    while (data_size != 0) 
+    long chunks_processed{};
+    while (data_size != 0)
     {
         const u32 chunk_size = data_size >= BLOCK_SIZE ? BLOCK_SIZE : data_size;
         sha_ctx.Update(data_pos, chunk_size);
         data_pos += chunk_size;
         data_size -= chunk_size;
+        if (yielder.has_value())
+        {
+            yielder.value()(chunks_processed);
+            ++chunks_processed;
+        }
     }
     m_buf.fill(0);
     sha_ctx.Final(m_buf.data());
     return { m_buf };
 }
 
+const inline xr_sha1::sha_checksum_t& xr_sha1::calculate(const u8* data, u32 data_size)
```

Comment:
What is the purpose of `inline` here? And why does this function exist to begin with, if one can just assign `yielder` to be `std::nullopt` or `nullptr`?

Reply:
Yes I have already thought about default parameter. I just missed this thing, yes, I will add it.
About `inline`: there is no need in this keyword here. I will remove it.